### PR TITLE
Redeem and Unstake after RevertStake

### LINKED ISSUE #738: Integration test: Progress redeem and unstake when revert redeem declared.
Revert and unstake can be progressed when the message status of outbox is `DeclaredRevocation` on the source chain.

### PR DESCRIPTION
**Pull request by abhayks1**
Fixes #738

### CODE REVIEW

**Review by jasonklein — DISMISSED** [review]
Mostly looks good to me. My comments are small and largely track the ones I made in #747.

 I see there are two different approaches taken in these test files: (a) `it`s for each major step in the process and (b) one `it` for the whole integrated flow. I prefer the integrated flow approach here, but I nevertheless note the inconsistency and think that a ticket to bring the test files in alignment would be 🌟(cc: @deepesh-kn).

**Also**: as discussed offline, this PR and #747 both introduce a file (and class), `revert_redeem_assertion.js` that are not identical—these changes should be aligned.

**Comment by abhayks1** [comment]
Working on the review feedback ⏩

File: test_integration/03_redeem_and_unstake/03_redeem_and_unstake_after_revert.js [hunk 1]
```diff
@@ -0,0 +1,405 @@
+// Copyright 2019 OpenST Ltd.
```
[the remark this answers is not in the document]
Comment (abhayks1) [reply test_integration/03_redeem_and_unstake/03_redeem_and_unstake_after_revert.js]: Agree 👍 
PR #747 also adds the file which start with 03.

File: test_integration/03_redeem_and_unstake/03_redeem_and_unstake_after_revert.js [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+    true,
+  );
+};
+
+describe('Redeem and Unstake after revert', async () => {
+  beforeEach(async () => {
+    originWeb3 = shared.origin.web3;
+    auxiliaryWeb3 = shared.auxiliary.web3;
+    originAccounts = shared.origin.accounts;
+    auxiliaryAccounts = shared.origin.accounts;
```
[the remark this answers is not in the document]
Comment (abhayks1) [reply test_integration/03_redeem_and_unstake/03_redeem_and_unstake_after_revert.js]: Good catch 😞 
Thank you for pointing 👍

File: test_integration/03_redeem_and_unstake/03_redeem_and_unstake_after_revert.js [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+    blockNumber,
+    messageStatus,
+    { from: redeemRequest.redeemer },
+  );
+  const transactionFeeInProgress = await getTransactionFee(
+    response,
+    auxiliaryWeb3,
+  );
+  const event = EventDecoder.getEvents(response, cogateway);
+  await progressRedeemAssertion.verify(
```
[the remark this answers is not in the document]
Comment (abhayks1) [reply test_integration/03_redeem_and_unstake/03_redeem_and_unstake_after_revert.js]: I feel yes 🍏 specifically in this test. 
Here we are testing that end to end redeem/unstake works fine even there is revertRedeem request in between the process.
Asserting progress redeem data points i.e. **balances** 🤑  assures full proof that the process completed successfully 💚.

**Comment by abhayks1** [comment]
Ready for review again 🚀

**Comment by abhayks1** [comment]
Resolving conflicts ⏰ 😿

**Comment by abhayks1** [comment]
Conflicts resolved ⛑ 
Ready for review again 💯

**Review by 0xsarvesh — APPROVED** [review]
LGTM 🎢

**Pull request merged**